test/vcf/optional_policy_test.cpp

```diff
@@ -21,7 +21,229 @@
+                                { vcf::PASS },
+                                { { "XYZ", "1" } },
+                                { },
+                                { },
+                                source})) );
+
+            CHECK_THROWS_AS( (optional_policy.optional_check_body_entry(parsing_state, vcf::Record{
+                                1,
+                                "chr1",
+                                0,
+                                { vcf::MISSING_VALUE },
+                                "A",
+                                { "AC" },
+                                1.0,
+                                { vcf::PASS },
+                                { { "XYZ", "1" } },
+                                { },
+                                { },
+                                source})),
+                            vcf::PositionBodyError*);
+        }
+    }
+
+    TEST_CASE("Id warnings", "[body id warnings]")
+    {
+        std::shared_ptr<vcf::Source> source{
+            new vcf::Source{
+                "Example VCF source",
+                vcf::InputFormat::VCF_FILE_VCF | vcf::InputFormat::VCF_FILE_BGZIP,
+                vcf::Version::v43}};
+
+        source->meta_entries.emplace(vcf::REFERENCE,
+            vcf::MetaEntry{
+                1,
+                vcf::REFERENCE,
+                "file",
+                source
+        });
+
+        source->meta_entries.emplace(vcf::CONTIG,
+            vcf::MetaEntry{
+                1,
+                vcf::CONTIG,
+                { { vcf::ID, "chr1" } },
+                source
+        });
+
+        source->meta_entries.emplace(vcf::INFO,
+            vcf::MetaEntry{
+                1,
+                vcf::INFO,
+                {
+                    { vcf::ID, "XYZ" },
+                    { vcf::NUMBER, "1" },
+                    { vcf::TYPE, vcf::INTEGER },
+                    { vcf::DESCRIPTION, "Random info tag" }
+                },
+                source
+        });
+
+        vcf::ParsingState parsing_state{source};
+
+        vcf::ValidateOptionalPolicy optional_policy;
+
+        SECTION("Commas in ID column")
+        {
+            CHECK_THROWS_AS( (optional_policy.optional_check_body_entry(parsing_state, vcf::Record{
+                                1,
+                                "chr1",
+                                123456,
+                                { "id1,id2" },
+                                "A",
+                                { "AC" },
+                                1.0,
+                                { vcf::PASS },
+                                { { "XYZ", "1" } },
+                                { },
+                                { },
+                                source})),
+                            vcf::IdBodyError*);
+        }
+    }
+
+    TEST_CASE("Record Reference Alternate matching warnings", "[body reference alternate matching warnings]")
```

Comment:
tags are a way to group tests, so it's not really useful to have so many specific tags.
https://github.com/catchorg/Catch2/blob/master/docs/test-cases-and-sections.md#tags
I would keep just `[warnings]` for these tests

Reply:
changed :)